src/screens/SearchResultsScreen.tsx

```diff
@@ -20,12 +23,39 @@ export async function loader({request}: {request: Request} ): Promise<string> {
  */
 export default function SearchResultsScreen() {
 	const query: string = useLoaderData() as string;
+	const location = useLocation();
+	const [search, setSearch] = useState(location.search.slice(2));
+	const [data, setData] = useState<MovieData | null>(null);
+	const [details, setDetails] = useState<MovieDetailsData | null>(null);
+
+	useEffect(() => {
+		console.log('hit');
+		const handler = async () => {
+			const movies = await getMoviesByName(search);
+			setData(movies);
+
+		};
+		handler();
+	}, []);
+
+	console.log(data);
+	useEffect(() => {
+		const handler = async () => {
+			const movies = await getMovieDetails(data!.results[0].id);
```

Comment:
is this supposed to be a `?` ? We could also wrap it in `if (data)`. Currently throwing a console error.

Reply:
The handler call is wrapped so that is resolved. I am still curious about this syntax though as I don't recall seeing it before.